internal/service/device.go

```diff
@@ -241,6 +241,7 @@ func (h *ServiceHandler) PatchDevice(ctx context.Context, request server.PatchDe
 	}
 
 	common.NilOutManagedObjectMetaProperties(&newObj.Metadata)
+	newObj.Metadata.ResourceVersion = nil
```

Comment:
Aren't we missing the point of resource version? Don't we want to avoid updating  if someone else updated the resource ?

Reply:
No, as I mentioned in the commit message, this is a PATCH.  The user didn't specify the resourceVersion - they just want to replace some field in the DB, regardless of what's currently there.  From the kubernetes docs:

> Instead of sending a PUT request, the client can send an instruction to the API server to patch an existing resource. A patch is typically appropriate if the change that the client wants to make isn't conditional on the existing data. Clients that need effective detection of lost updates should consider making their request conditional on the existing resourceVersion (either HTTP PUT or HTTP PATCH), and then handle any retries that are needed in case there is a conflict.

Reply:
OK